### PR TITLE
Add red hat distro support for vscode extension use package searching

### PR DESCRIPTION
**Pull request by JL03-Yue**

### CODE REVIEW

File: vscode-dotnet-runtime-library/distro-data/distro-support.json [hunk 1]
```diff
@@ -35,6 +35,14 @@
 				"commandParts": ["upgrade", "-y", "{packageName}"]
 			}
 		],
+		"searchCommand":
+		[
+			{
+				"runUnderSudo": false,
+				"commandRoot": "apt-cache",
+				"commandParts": ["search", "{0}"]
```
Comment (JL03-Yue) [inline vscode-dotnet-runtime-library/distro-data/distro-support.json:43]: Replace {0} with packageName

File: vscode-dotnet-runtime-library/distro-data/distro-support.json [hunk 2]
```diff
@@ -164,37 +179,153 @@
...
+				]
+			}
+		],
+		"searchCommand": [
+			{
+				"runUnderSudo": false,
+				"commandRoot": "yum",
+				"commandParts": [
+					"search",
+					"{0}",
```
Comment (JL03-Yue) [inline vscode-dotnet-runtime-library/distro-data/distro-support.json:229]: Replace {0} with packageName

File: vscode-dotnet-runtime-library/distro-data/distro-support.json [hunk 3]
```diff
@@ -164,37 +179,153 @@
...
+				"commandRoot": "yum",
+				"commandParts": [
+					"list",
+					"install",
+					"{packageName}",
+					"-q"
+				]
+			}
+		],
+		"expectedDistroFeedInstallDirectory": "/usr/lib64/dotnet/sdk",
```
[the remark this answers is not in the document]
Comment (JL03-Yue) [reply vscode-dotnet-runtime-library/distro-data/distro-support.json:258]: From their [documentation](https://access.redhat.com/articles/rhel-limits), red hat 7 and newer only support x86_64 not 32 bit. [RedHat 8](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/8.0_release_notes/architectures) and [RedHat 9](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/9.0_release_notes/architectures)
supports AMD64, Intel 64, Arm 64. IBM Z 64, no 32 bits archs

Comment (nagilson) [inline vscode-dotnet-runtime-library/distro-data/distro-support.json:258]:
```suggestion
		"expectedDistroFeedInstallDirectory": "/usr/lib64/dotnet/dotnet",
```

Comment (nagilson) [reply vscode-dotnet-runtime-library/distro-data/distro-support.json:258]: We just made a modification to the API so it returns the path to the dotnet object file to execute, so this should also be changed here

File: vscode-dotnet-runtime-library/src/Acquisition/RedHatDistroSDKProvider.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public async getInstalledGlobalDotnetPathIfExists(installType : LinuxInstallType) : Promise<string | null>
+    {
+        this.commandRunner.returnStatus = true;
+        const commandResult = await this.commandRunner.executeMultipleCommands(this.myDistroCommands(this.currentInstallPathCommandKey));
+        this.commandRunner.returnStatus = false;
+
+        if (commandResult[0] !== '0'){
+            return ''
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/src/Acquisition/RedHatDistroSDKProvider.ts:27]:
```suggestion
            return '';
```

File: vscode-dotnet-runtime-library/src/test/unit/RedHatDistroTests.test.ts [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            const status = await provider.getDotnetVersionSupportStatus(mockVersion, installType);
+            assert.equal(status, DotnetDistroSupportStatus.Distro);
+        }
+    }).timeout(standardTimeoutTime);
+
+    test('Gets Distro Feed Install Dir', async () => {
+        if(shouldRun)
+        {
+            const distroFeedDir = await provider.getExpectedDotnetDistroFeedInstallationDirectory();
+            assert.equal(distroFeedDir, '/usr/lib64/dotnet/sdk');
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/src/test/unit/RedHatDistroTests.test.ts:55]:
```suggestion
            assert.equal(distroFeedDir, '/usr/lib64/dotnet/dotnet');
```

File: vscode-dotnet-runtime-library/src/Acquisition/LinuxVersionResolver.ts [hunk 6]
```diff
@@ -69,6 +71,7 @@ export class LinuxVersionResolver
     If you would like to proceed with installing .NET automatically for VS Code, you must remove this custom installation.
     Your custom install is located at: `;
     public baseUnsupportedDistroErrorMessage = 'We are unable to detect the distro or version of your machine';
+    public redhatUnsupportedDistroErrorMessage = 'Red Hat Entreprise Linux 7.0 is currently not supported. Please install Red Hat Enterprise Linux 8.0 or Red Hat Enterprise Linux 9.0 from https://access.redhat.com/downloads/';
```
Comment (nagilson) [inline vscode-dotnet-runtime-library/src/Acquisition/LinuxVersionResolver.ts:74]:
```suggestion
    public redhatUnsupportedDistroErrorMessage = 'Red Hat Entreprise Linux 7.0 is currently not supported. Follow the instructions here to download the .NET SDK: https://learn.microsoft.com/en-us/dotnet/core/install/linux-rhel#rhel-7--net-6. Or, install Red Hat Enterprise Linux 8.0 or Red Hat Enterprise Linux 9.0 from https://access.redhat.com/downloads/';
```

**Review by nagilson — COMMENTED** [review]
Thank you for including a test for the RHEL 7 notice. Overall it looks great to me, lets test this with my new changes together and if it works well we can merge it! 🥳

**Pull request merged**